src/mds/SessionMap.cc

```diff
@@ -924,6 +983,35 @@ int Session::check_access(CInode *in, unsigned mask,
   return 0;
 }
 
+void SessionMap::hit_session(Session *session) {
+  auto it = session_load_avg_map.find(session->info.inst.name);
+  assert(it != session_load_avg_map.end());
+
+  auto &session_load_avg = it->second;
+  session_load_avg.adjust();
+
+  // recalculate total load and average across sessions
+  double total_load = 0.0;
+  double avg_load;
+
+  int sessions = 0;
+  for (auto const &it : session_load_avg_map) {
+    // TODO: use session as map key (maybe?)
+    auto s_it = session_map.find(it.first);
```

Comment:
I'm dubious on the value of the `session_load_avg_map` since it will not be necessary to actually lookup the DecayCounter for each map. Let's just put the `DecayCounter` in the `Session` and just iterate the `session_map`?

Reply:
Makes sense as you suggested below.